src/components/DatePickerInline/DatePickerInlineBase.tsx

```diff
@@ -15,37 +15,70 @@ import type {
     MultiChange,
     CalendarDates,
 } from './types';
+import { createFastContext } from '../../fast-context';
+import MonthPicker from './MonthPicker';
+import DatePickerDockedHeader from '../DatePickerDocked/DatePickerDockedHeader';
+
+export type Store = {
+    localDate: Date;
+    startDateYear: number;
+    endDateYear: number;
+    pickerType: 'month' | 'year' | undefined;
+};
+
+const defaultValue = {
+    localDate: new Date(),
+    startDateYear: 1800,
+    endDateYear: 2200,
+    pickerType: undefined,
+};
+
+const { Provider, useSelector } = createFastContext<Store>(defaultValue);
 
 function DatePickerInlineBase(props: DatePickerInlineBaseProps) {
+    return (
+        <Provider value={defaultValue}>
+            <DatePickerInlineComponent {...props} />
+        </Provider>
+    );
+}
+
+function DatePickerInlineBaseChild(props: DatePickerInlineBaseProps) {
     const {
         locale = 'en',
         mode = 'single',
         onChange,
+        startYear = 1800,
+        endYear = 2200,
         startDate,
         endDate,
         date,
         disableWeekDays,
-        startYear,
-        endYear,
         dates,
         validRange,
         dateMode,
         style,
+        isDocked,
```

Comment:
it's not required.
Pass down the header from the parent as a Component or as a ReactNode (either of them work)
wherever else isDocked is required, put the specific values in context from he parent component, or use necessary props for it.

Reply:
removed @shashank1010